### ISSUE TITLE
Command not found

### ISSUE THREAD

**Issue by rapha8l** [jheysonsaav/catsh#1]
Hi,

On Void linux x86_64, I  get
```
./target/release/stellar 
Error: Command not found
```

And two warnings at build
```

warning: associated function is never used: `cache_dir`
  --> src/utils/dirs.rs:84:12
   |
84 |     pub fn cache_dir(&self) -> &Path {
   |            ^^^^^^^^^
   |
   = note: `#[warn(dead_code)]` on by default

warning: associated function is never used: `data_dir`
  --> src/utils/dirs.rs:88:12
   |
88 |     pub fn data_dir(&self) -> &Path {
   |            ^^^^^^^^

warning: 2 warnings emitted

```

Thanks

**Comment by jheysonsaav** [comment]
hello @rapha8l, my idea with the project is to create an api so that plugins can be created in an easy way and be able to integrate them with the shell, I plan to make a plugin called "base" which will have the basic commands and will be installed by default, sorry for not explaining it in the readme 😅

**Comment by jheysonsaav** [comment]
I also refactored the code a bit to avoid this type of warnings and added clippy to the github actions workflow

**Comment by rapha8l** [comment]
Thanks
I pulled and tried a build release and got this error

```
error[E0583]: file not found for module `settings`
 --> core/lib.rs:4:1
  |
4 | pub mod settings;
  | ^^^^^^^^^^^^^^^^^
  |
  = help: to create the module `settings`, create file "core/settings.rs"

error: aborting due to previous error

For more information about this error, try `rustc --explain E0583`.
error: could not compile `catsh-core`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

**Comment by jheysonsaav** [comment]
@rapha8l I have already fixed that little bug and have released a new release [v0.1.0-beta.0](https://github.com/jheysonsaav/catsh/releases/tag/v0.1.0-beta.0)

Even the shell is not working and it cannot be used day by day so I have not yet released a stable version, I am working so that the next few days it can be used well and be able to release a stable version

**Comment by rapha8l** [comment]
thanks !
